src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs

```diff
@@ -133,38 +133,133 @@ private void EmbedIfNeedTo(BoundExpression receiver, ImmutableArray<PropertySymb
             }
         }
 
+        public void InterceptCallAndAdjustArguments(
+            ref MethodSymbol method,
+            ref BoundExpression? receiverOpt,
+            ref ImmutableArray<BoundExpression> arguments,
+            ref ImmutableArray<RefKind> argumentRefKindsOpt,
+            ref bool invokedAsExtensionMethod,
+            Location? interceptableLocation)
+        {
+            // PROTOTYPE(ic):
+            // Add assertions for the possible shapes of calls which could come through this method.
+            // When the BoundCall shape changes in the future, force developer to decide what to do here.
+
+            if (this._compilation.GetInterceptor(interceptableLocation) is not var (interceptsLocationAttributeData, interceptor))
```

Comment:
nit: maybe `TryGet...` pattern would be more suitable/direct. It could avoid the assertion below (line 154) with appropriate nullability attribute.

Reply:
Adding a comment and will address in follow-up PR which will handle duplicates, etc.